tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
+
+        duplicate_df["unique_combinations_for_test"] = (
+            duplicate_df["PortGroupCode"]
+            + "-"
+            + duplicate_df["FundCode"]
+            + "-"
+            + duplicate_df["Scope"]
+        )
+
+        self.assertEqual(
+            len(
+                [
+                    port_group
+                    for nested_group in [
+                        port_group.portfolios
+                        for port_group in responses["portfolio_groups"]["success"]
+                    ]
+                    for port_group in nested_group
+                ]
+            ),
+            len(duplicate_df["unique_combinations_for_test"].unique()),
+        )
+
+    def test_port_group_update(self):
+
+        """
+        Description:
+        ------------
+        Here we test updating a portfolio group which already exists.
+        """
+
+        update_df = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_group_update.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=update_df,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
+
+        first_group_series = update_df.iloc[0]
+        first_group_resoruce_id = lusid.models.ResourceId(
+            scope=first_group_series["Scope"], code=first_group_series["FundCode"]
+        )
+
+        get_first_group_response_portfolios = (
+            self.api_factory.build(lusid.api.PortfolioGroupsApi)
+            .get_portfolio_group(
+                scope=self.scope, code=first_group_series["PortGroupCode"]
+            )
+            .portfolios
+        )
+
+        self.assertIn(first_group_resoruce_id, get_first_group_response_portfolios)
+        self.assertNotIn(
+            lusid.models.ResourceId(scope="test_scope_20200225", code="TEST_COM5"),
+            get_first_group_response_portfolios,
+        )
```

Comment:
What is the significance of checking that this portfolio does not exist in the group?

Reply:
You are missing a test for updating a portfolio group when a portfolio that you are updating with does not exist. Please document the behaviour for this with a test.

Reply:
(1) On second look, it's a weak test - I will remove. 
(2) There should be a test called `test_portfolio_not_exist` for that.